### PR TITLE
Fix: Screen reader does not announce selected date, current month

### PR DESCRIPTION
**Pull request by SpaNb4**
This PR fixes some accessibility issues, and may help screen reader users navigate this datepicker library better.

- With my fix when you change the month, screen reader will announce the current month. Because with current implementation, when clicking on the next/previous month button, the user does not know which month the calendar is currently on
- Also made some changes so that screenreader will announce selected date

I believe that my PR will resolve these issue: 
https://github.com/Hacker0x01/react-datepicker/issues/3558
https://github.com/Hacker0x01/react-datepicker/issues/3559 
https://github.com/Hacker0x01/react-datepicker/issues/3766

### CODE REVIEW

**Comment by SpaNb4** [comment]
@martijnrusschen Do I need to add more tests to increase the code-coverage and only then will you review this PR, right?

File: src/index.jsx [hunk 1]
```diff
@@ -997,6 +1003,75 @@ export default class DatePicker extends React.Component {
...
+          ? "End date: " +
+            safeDateFormat(this.props.endDate, {
+              dateFormat: longDateFormat,
+              locale,
+            })
+          : ""
+      }`;
+    } else {
+      if (this.props.showTimeSelectOnly) {
+        ariaLiveMessage = `Selected time: ${safeDateFormat(
```
Comment (ch1qui) [inline src/index.jsx:1031]: Hardcoded english texts will ruin accessibility for all non-english users. This prefix should be diffferent prop.

**Comment by martijnrusschen** [comment]
Can you resolve the merge conlfict so we can get this merged and released?

File: src/calendar.jsx [hunk 2]
```diff
@@ -1004,6 +1040,7 @@ export default class Calendar extends React.Component {
           showPopperArrow={this.props.showPopperArrow}
           arrowProps={this.props.arrowProps}
         >
+          {this.renderAriaLiveRegion()}
```
Comment (SergioVanacloigCarlsberg) [inline src/calendar.jsx:1043]: Shouldn't this span be optional?

Comment (LukeBrandon) [reply src/calendar.jsx:1043]: I agree, this should be optional.  This is showing up in my application and I cannot opt out of showing the "Selected date:" text.

**Pull request merged**